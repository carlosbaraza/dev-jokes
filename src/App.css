.App {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100vh;
  color: #aaffaa;
  background: #222;
  font-family: monospace;
}

.App h2:after {
  content: "\2588";
  animation: 1s fade infinite;
}

@keyframes fade {
  0% {
    opacity: 0;
  }
  50% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}
