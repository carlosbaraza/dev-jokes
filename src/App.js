import React, { Component } from "react";
import logo from "./logo.svg";
import "./App.css";
import jokes from "./jokes.json";

const sortedJokes = jokes.sort(
  (j1, j2) => (j1.createdAt < j2.createdAt ? 1 : -1)
);

const joke = sortedJokes[0];

class App extends Component {
  render() {
    return (
      <div className="App">
        <h2
          dangerouslySetInnerHTML={{
            __html: joke.body
          }}
        />
      </div>
    );
  }
}

export default App;
