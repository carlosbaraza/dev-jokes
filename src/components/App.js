import React, { Component } from "react";
import "./App.css";
import { ClapIcon } from "./ClapIcon";
import { fetchJoke } from "../services/jokes";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      joke: fetchJoke(0),
      jokeIndex: 0
    };
  }

  nextJoke = () => {
    const joke = fetchJoke(this.state.jokeIndex + 1);

    if (joke) {
      this.setState({
        joke,
        jokeIndex: this.state.jokeIndex + 1
      });
    }
  };

  render() {
    const { joke } = this.state;

    return (
      <div className="App">
        <h2
          dangerouslySetInnerHTML={{
            __html: joke.body
          }}
        />
        <div className="controls">
          {/* <div className="claps">
            <ClapIcon className="claps__ClapIcon controls__button" />
            <p>{joke.claps} claps</p>
          </div> */}

          <div className="moar controls__button" onClick={this.nextJoke}>
            MOAR!
          </div>
        </div>
      </div>
    );
  }
}

export default App;
