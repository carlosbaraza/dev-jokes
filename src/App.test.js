import React from "react";
import App from "./App";
import { shallow } from "enzyme";

it("renders the joke app", () => {
  expect(shallow(<App />)).toMatchSnapshot();
});
