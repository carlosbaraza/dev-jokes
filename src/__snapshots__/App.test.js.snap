// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the joke app 1`] = `
<div
  className="App"
>
  <h2>
    Programmers count from 0...
  </h2>
</div>
`;
